Modules/socketmodule.c

```diff
@@ -2273,7 +2273,9 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
         switch (s->sock_proto) {
 #ifdef CAN_RAW
         case CAN_RAW:
+        #if !defined(__NetBSD__)
```

Comment:
Does NetBSD provide CAN_BCM? If not, I would suggest to replace `#if !defined(__NetBSD__)` with `#ifndef CAN_BCM`.

Reply:
I think we should use `#ifdef CAN_BCM` rather than `#ifndef CAN_BCM`. Using `#ifndef CAN_BCM` would enable the fallthrough annotation on systems where CAN_BCM is not defined. 
However, on NetBSD, where `CAN_BCM` is not defined, the presence of the fallthrough annotation causes the compiler to generate an error because there is no next case to fall through to. By using `#ifdef CAN_BCM`, we ensure the fallthrough annotation only appears when the next case actually exists.

Reply:
Ah, right